### PR TITLE
Allinea-forge: Add support for Ubuntu distro

### PR DESCRIPTION
**Pull request by jorblancoa**
- Created getDistroName() function
- Added Ubuntu version 16.04

### CODE REVIEW

File: var/spack/repos/builtin/packages/allinea-forge/package.py [hunk 1]
```diff
@@ -4,7 +4,16 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+import platform
 
+def getDistroName():
+    distro = platform.linux_distribution()
```
Comment (tristan0x) [inline var/spack/repos/builtin/packages/allinea-forge/package.py:10]: On OSX:

```python
>>> platform.linux_distribution()
('', '', '')
```

File: var/spack/repos/builtin/packages/allinea-forge/package.py [hunk 2]
```diff
@@ -16,6 +25,7 @@ class AllineaForge(Package):
 
     version('6.0', 'c85fec6d01680b5b46fea80111186244')
     version('7.0', '4c2f5b2ff83d494854df74e6df4be7be')
+    version('16.04', 'd8396b046e9b7f4241d69466f6155790')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/allinea-forge/package.py]: This is messed up. @pramodk added versions here. Correct would be `latest`, and at most variants for the distro? IMO, no variants needed actually.

File: var/spack/repos/builtin/packages/allinea-forge/package.py [hunk 3]
```diff
@@ -26,9 +36,10 @@ class AllineaForge(Package):
 
     def url_for_version(self, version):
         # TODO: add support for other architectures/distributions
+        distro = getDistroName()
         url = "http://content.allinea.com/downloads/"
-        return url + "arm-forge-latest-Redhat-%s-x86_64.tar" % version
-
+        return url + "arm-forge-latest-%s-%s-x86_64.tar" % (distro, version)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/allinea-forge/package.py]: Allinea's version is `latest`! `version` as used here points to the distribution version, and should also be programatically determined.

FWIW, by getting different URLs for the same `version` (in Spack terminology), we're messing with the source hash for mirroring. Not important here, just a future note/warning.

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by matz-e** [comment]
@jorblancoa I think that @pramodk did the current versioning scheme to avoid checksum conflicts. I've solved this [here](https://github.com/spack/spack/pull/10024/files#diff-cbbfc6835dea07fbf9ed98ee676aa447R33) by setting the right checksum based on the tarball downloaded.

I think your distro + distro-version detection combined with one `latest` version should be feasible, even though we loose the ability to mirror the tarballs (one tarball per version mirrored).

**Comment by jorblancoa** [comment]
I removed the distro versions and added the package real versions for both redhat and ubuntu.

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/allinea-forge/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/allinea-forge/package.py:10]: this package is only for linux, added conflict

**Comment by pramodk** [comment]
thank you @jorblancoa !

**Pull request merged**